internal/oauth2/flow_client_credentials.go

```diff
@@ -26,6 +28,18 @@ type clientCredentialsConfigurator interface {
 	fositex.SigningKeyProvider
 }
 
+func buildClientURN(c fosite.Client) string {
+	clientUUID := uuid.MustParse(c.GetID())
+
+	urn, err := urnx.Build(types.URNNamespace, types.URNResourceTypeUser, clientUUID)
+	if err != nil {
+		// If for some reason we aren't building valid URNs, panic
+		panic(err)
```

Comment:
why not bubble this up and return an error to the handler?

Reply:
I'm kind of going back and forth on this. My stance here is mostly that we should never be in a position where we're generating non-UUID identifiers, and if we are then something is seriously wrong. I can update this to use errors instead though if that works better for you; we panic elsewhere though for the same reason.